File: code/java/setZeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        Set<Integer> cols = new HashSet();
        Set<Integer> rows = new HashSet();
        for(int i=0;i<matrix.length;i++)
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j] == 0){
                    if(!rows.contains(j))
                        rows.add(j);
                    if(!cols.contains(i))
                        cols.add(i);
                }
            }
        for(int i=0;i<matrix.length;i++)
            for(int j=0;j<matrix[0].length;j++)
                if(rows.contains(j) || cols.contains(i))
                    matrix[i][j] = 0;
    }
}

File: code/python/setZeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        cols = set()
        rows = set()
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == 0:
                    if i not in cols:
                        cols.add(i)
                    if j not in rows:
                        rows.add(j)
        
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if(i in cols) or (j in rows):
                    matrix[i][j] = 0
        
